### ISSUE TITLE
Evaluate Jupyter notebooks for this project

### ISSUE THREAD

**Issue by bobturneruk** [ScottishCovidResponse/SCRCIssueTracking#110]
Evaluate Jupyter notebooks as a means of documenting / disseminating this project. Will compute requirements restrict this?

**Comment by alex-konovalov** [comment]
I believe that Jupyter notebooks may have a strong potential for disseminating the project. For documenting, Jupyter-based tutorials may also be useful. I put together a template for publishing reproducible experiments in Python Jupyter notebooks runnable on Binder and Azure Notebooks some time ago - please see here: https://github.com/rse-standrewscs/python-binder-template (it does not have Codecov integration yet - that's in my TODOs). You can click on the "binder" button to start the project on Binder, and then find the notebook in the `notebooks` directory. You can also use RISE extension to view a reveal.js-like slideshow, but with Jupyter, where all code cells are live!

So for example, in https://github.com/ScottishCovidResponse/simple_network_sim/ one could have a directory for notebooks which load and demonstrate the code. As a newcomer to this project, I don't know what are the computational requirements, but I can envisage that interested users may first try to run sample notebooks on Binder, and then clone the repository and run code locally if they wish to proceed.

**Comment by alex-konovalov** [comment]
I am now looking for a small example for a proof of concept demo. Is https://github.com/ScottishCovidResponse/simple_network_sim/blob/master/simple_network_sim/sampleUseOfModel.py good for this? Are there any usage instructions?

**Comment by bobturneruk** [comment]
Thanks for looking at this @alex-konovalov. I think it's probably best to focus on the priority / starter issues for now, if that's OK? The usefulness of Jupyter notebooks is going to depend a lot on how the results of this project are to be disseminated. Do feel free to self-assign and continue on this issue, if you want. All input is appreciated, however it may be more difficult to draw on other team members.

**Comment by alex-konovalov** [comment]
@bobturneruk thanks, that's OK! I've also found out that the usage instructions for this file are in  the README. It would be useful irrespectively to this issue to add a comment to that file to document its usage, in particular, the meaning of the command line arguments.

**Comment by bobturneruk** [comment]
Good point, appended to https://github.com/ScottishCovidResponse/SCRCIssueTracking/issues/257

**Comment by bobturneruk** [comment]
Visualisation will be handled by the visualisation team, so any use of Jupyter would be for development and testing purposes.

**Comment by FiodorG** [comment]
Not sure if this is any help for this issue, but I've setup a Jupyter notebook to help me run the parameter sensitivity analysis (https://github.com/ScottishCovidResponse/simple_network_sim/blob/parameter-sensitivity-analysis/Parameter%20sensitivity%20analysis.ipynb). There are no issues as far as I can see and it's very handy.

**Comment by alex-konovalov** [comment]
Thanks @FiodorG - it's good to have a runnable notebooks to start with, so this is helpful indeed!
